src/transformers/models/emformer/modeling_emformer.py

```diff
@@ -0,0 +1,1158 @@
+        frames; S: number of summary elements; M: number of memory elements.
+
+        Args:
+            utterance (torch.Tensor): utterance frames, with shape *(T, B, D)*.
+            lengths (torch.Tensor): with shape *(B,)* and i-th element representing
+                number of valid frames for i-th batch element in `utterance`.
+            right_context (torch.Tensor): right context frames, with shape *(R, B, D)*.
+            summary (torch.Tensor): summary elements, with shape *(S, B, D)*.
+            mems (torch.Tensor): memory elements, with shape *(M, B, D)*.
+            attention_mask (torch.Tensor): attention mask for underlying attention module.
+
+        Returns:
+            (Tensor, Tensor):
+                Tensor
+                    output frames corresponding to utterance and right_context, with shape *(T + R, B, D)*.
+                Tensor
+                    updated memory elements, with shape *(M, B, D)*.
+        """
+        output, output_mems, _, _, attention_probs = self._forward_impl(
+            utterance, lengths, right_context, summary, mems, attention_mask, output_attentions
+        )
+        return output, output_mems[:-1], attention_probs
+
+
+class EmformerLayer(torch.nn.Module):
+    r"""Emformer layer that constitutes Emformer.
+
+    Args:
+        input_dim (int): input dimension.
+        num_heads (int): number of attention heads.
+        ffn_dim: (int): hidden layer dimension of feedforward network.
+        segment_length (int): length of each input segment.
+        dropout (float, optional): dropout probability. (Default: 0.0)
+        activation (str, optional): activation function to use in feedforward network.
+            Must be one of ("relu", "gelu", "silu"). (Default: "relu")
+        left_context_length (int, optional): length of left context. (Default: 0)
+        max_memory_size (int, optional): maximum number of memory elements to use. (Default: 0)
+        tanh_on_mem (bool, optional): if `True`, applies tanh to memory elements. (Default: `False`)
+        negative_inf (float, optional): value to use for negative infinity in attention weights. (Default: -1e8)
+    """
+
+    def __init__(
+        self,
+        input_dim: int,
+        num_heads: int,
+        ffn_dim: int,
+        segment_length: int,
+        dropout: float = 0.0,
+        activation: str = "relu",
+        left_context_length: int = 0,
+        max_memory_size: int = 0,
+        tanh_on_mem: bool = False,
+        negative_inf: float = -1e8,
+    ):
+        super().__init__()
+
+        self.attention = EmformerAttention(
+            input_dim=input_dim,
+            num_heads=num_heads,
+            dropout=dropout,
+            tanh_on_mem=tanh_on_mem,
+            negative_inf=negative_inf,
+        )
+        self.dropout = torch.nn.Dropout(dropout)
+
+        activation_module = _get_activation_module(activation)
```

Comment:
Would be in favor of just using the function defined in our `activations` module